packages/@aws-cdk/aws-glue/lib/job.ts

```diff
@@ -0,0 +1,872 @@
+   * @param options optional event options
+   * @private
+   */
+  private rule(jobState: JobState, id?: string, options: events.OnEventOptions = {}): events.Rule {
+    // No caching
+    if (id) {
+      const rule = this.onEvent(id, options);
+      rule.addEventPattern({
+        detail: {
+          state: [jobState],
+        },
+      });
+      return rule;
+    }
+    // Caching
+    const ruleId = `${jobState}Rule`;
+    if (!this.cachedRules[ruleId]) {
+      const rule = this.onEvent(ruleId, {
+        description: `Rule triggered when Glue job ${this.jobName} is in ${jobState} state`,
+      });
+      rule.addEventPattern({
+        detail: {
+          state: [jobState],
+        },
+      });
+      this.cachedRules[ruleId] = rule;
+    }
+    return this.cachedRules[ruleId];
+  }
+
+}
+
+/**
+ * Attributes for importing {@link Job}.
+ */
+export interface JobAttributes {
+  /**
+   * The name of the job.
+   */
+  readonly jobName: string;
+}
+
+/**
+ * Construction properties for {@link Job}.
+ */
+export interface JobProps {
+  /**
+   * The name of the job.
+   *
+   * @default cloudformation generated name.
+   */
+  readonly jobName?: string;
+
+  /**
+   * The description of the job.
+   *
+   * @default no value.
+   */
+  readonly description?: string;
+
+  /**
+   * The number of AWS Glue data processing units (DPUs) that can be allocated when this job runs.
+   * Should not be used for Glue version 2.0 and later - workerType and numberOfWorkers should be used instead.
+   *
+   * @default 10 when you specify an Apache Spark ETL or Sreaming job, 0.0625 DPU when you specify a Python shell job.
+   */
+  readonly maxCapacity?: number;
+
+  /**
+   * The maximum number of times to retry this job after a JobRun fails.
+   *
+   * @default 0
+   */
+  readonly maxRetries?: number;
+
+  /**
+   * The maximum number of concurrent runs allowed for the job.
+   * An error is returned when this threshold is reached. The maximum value you can specify is controlled by a service limit.
+   *
+   * @default 1.
+   */
+  readonly maxConcurrentRuns?: number;
+
+  /**
+   * The number of minutes to wait after a job run starts, before sending a job run delay notification.
+   *
+   * @default N/A (no delay notifications)
+   */
+  readonly notifyDelayAfter?: cdk.Duration;
+
+  /**
+   * The maximum time that a job run can consume resources before it is terminated and enters TIMEOUT status.
+   *
+   * @default 2,880 minutes (48 hours)
+   */
+  readonly timeout?: cdk.Duration;
+
+  /**
+   * Glue version determines the versions of Apache Spark and Python that AWS Glue supports. The Python version indicates the version supported for jobs of type Spark.
+   *
+   */
+  readonly glueVersion: GlueVersion;
+
+  /**
+   * The type of predefined worker that is allocated when a job runs.
+   * Should not be used for Glue version 1.0 and earlier - maxCapacity should be used instead.
+   *
+   * @default differs based on specific glueVersion
```

Comment:
let's describe this if possible

Reply:
I sincerely don't know!

Reply:
okay, please do some testing to find out (deploy a bunch of versions without these set and see what Glue does)